src/provable/base58.ts

```diff
@@ -0,0 +1,66 @@
+import { versionBytes } from '../js_crypto/constants.js';
+import { Ledger } from '../snarky.js';
+import { Binable, withVersionNumber } from './binable.js';
+
+export { base58, fieldEncodings, Base58 };
+
+type Base58<T> = {
+  toBase58(t: T): string;
+  fromBase58(base58: string): T;
+};
+
+function base58<T>(binable: Binable<T>, versionByte: number): Base58<T> {
+  return {
+    toBase58(t) {
+      let bytes = binable.toBytes(t);
+      let binaryString = String.fromCharCode(...bytes);
+      let ocamlBytes = { t: 9, c: binaryString, l: bytes.length };
```

Comment:
What's the significance of the `9` value here? Could we add a comment to explain why this number is specifically this value? If this was changed, what's the downstream effects? :O

Reply:
I added a comment!

Reply:
It's a JSOO thing